scripts/cluster-install/install.py

```diff
@@ -0,0 +1,392 @@
+        config_file['manager']['hostname'] = node.name
+        config_file['manager']['cloudify_license_path'] = \
+            REMOTE_INSTALL_CLUSTER + '/license.yaml'
+        config_file['rabbitmq']['cluster_members'] = rabbitmq_members
+        config_file['rabbitmq']['username'] = rabbitmq_credentials[0]
+        config_file['rabbitmq']['password'] = rabbitmq_credentials[1]
+        config_file['rabbitmq']['ca_path'] = ca_path
+        config_file['postgresql_server']['cluster']['nodes'] = postgresql_nodes
+        config_file['postgresql_server']['ca_path'] = ca_path
+        config_file['ssl_inputs'] = _create_ssl_inputs(node.name)
+        suffix = '/config_files/{}_config.yaml'.format(node.name)
+        conf_file_name = LOCAL_INSTALL_CLUSTER + suffix
+        with open(conf_file_name, 'w') as f:
+            yaml.dump(config_file, f)
+
+
+def _prepare_config_files(instances_dict):
+    logging.info('Preparing config files')
+    os.mkdir(LOCAL_INSTALL_CLUSTER + '/config_files')
+    _prepare_postgresql_config_files(instances_dict)
+    rabbitmq_credentials = _prepare_rabbitmq_config_files(instances_dict)
+    _prepare_manager_config_files(instances_dict, rabbitmq_credentials)
+
+
+def _download_manager_and_create_license(license_path, download_link):
+    os.system('cd {0} && curl -O {1}'.format(LOCAL_INSTALL_CLUSTER,
+                                             download_link))
+    os.system('cp {0} {1}'.format(
+        license_path, os.path.join(LOCAL_INSTALL_CLUSTER, 'license.yaml')))
+
+
+def _install_instances(instances_dict, key_path, rpm_name):
+    logging.info('Installing instances')
+    instances_list = _create_instances_list(instances_dict,
+                                            include_load_balancer=False)
+    for instance in instances_list:
+        logging.info('Installing {}'.format(instance.name))
+        scp_local_to_remote(key_path, instance, LOCAL_INSTALL_CLUSTER,
+                            REMOTE_PARENT_DIRECTORY)
+        instance.exec_command('sudo yum install -y {rpm}'.format(
+            rpm=os.path.join(REMOTE_INSTALL_CLUSTER, rpm_name)))
+        instance.exec_command('cp {0}/config_files/{1}_config.yaml '
+                              '/etc/cloudify/config.yaml'.
+                              format(REMOTE_INSTALL_CLUSTER, instance.name))
+        install_command = 'cfy_manager install --private-ip {0} --public-ip ' \
+                          '{1}'.format(instance.private_ip, instance.public_ip)
+        stdin, stdout, stderr = instance.exec_command(install_command)
+        print(stdout.read())
+        time.sleep(0.5)  # Avoiding running over the print by the next logging
+
+
+def _get_vm(instances_details, instance, key_path, vm_username):
+    private_ip = str(instances_details[instance][0])
+    public_ip = str(instances_details[instance][1])
+    return VM(private_ip, public_ip, instance, key_path, vm_username)
+
+
+def _get_instances_dict(instances_details, key_path, vm_username):
```

Comment:
i'd consider making a `Cluster` class (with an `__iter__` for all instances, and also `.manager_nodes` etc properties) to hold those instances, because currently we're passing this sometimes as a dict, then making it a list, etc

Reply:
Great point, I'll work on it in future improvements.